### PR TITLE
Reduction scheduler refactor

### PR DESCRIPTION
**Pull request by csarofeen**
Clean up associated with scheduling reduction tv. This also expands the reduction scheduler with grid dimensions on reductions for normalization kernels in preparation for cross-block normalization persistence.

PR Includes:
Renaming in reduction heuristics
Cleanup/renaming in reduction/normalization heuristics calculation
More consistent runtime setting of block/thread launch parameters
Minor fix in persistent buffer calculation
Minor fix in vectorization

Relies on:
https://github.com/csarofeen/pytorch/pull/1296
https://github.com/csarofeen/pytorch/pull/1297
https://github.com/csarofeen/pytorch/pull/1298

### CODE REVIEW

**Comment by naoyam** [comment]
All Python tests, including `TestCudaFuser.test_native_layer_norm_half`, are passing. Deprecating #1280.

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp [hunk 1]
```diff
@@ -534,9 +538,9 @@ ReductionParams OuterReductionHeuristic(
   // domain for this
 
   // Blocks for reductions
-  int64_t gdimy = 1;
+  int64_t grdim = 1;
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp:541]: What do the names of `grdim` and `gidim` mean?

**Review by naoyam — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp:541]: grid dim in the reduction and grid dim in the iter, will add a comment.

**Pull request merged**